chore: add Js string methods transcript (#58897)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lecture-working-with-common-string-methods/67326c0d7bef01c539120766.md

# --description--

Watch the video or read the transcript and answer the questions below.

# --transcript--

How can you test if a string contains a specific substring in JavaScript?

When working with strings in JavaScript, there are many cases where you might need to check whether a string contains a specific substring, which is a smaller part of that string.

For example, you might want to check if a user's input includes a specific word or character before performing some action. One way to achieve this is by using the `includes()` method.

The `includes()` method is used to check if a string contains a specific substring. If the substring is found within the string, the method returns `true` otherwise, it returns `false`.

Here's the basic syntax:

```js
string.includes(searchValue);
```

> For the syntax, the `searchValue` is the substring you want to look for within the string.
> 
> Here’s a example:

Reviewer: ```suggestion
For the syntax, the `searchValue` is the substring you want to look for within the string. And here's a example:
```

```js
let phrase = "JavaScript is awesome!";
let result = phrase.includes("awesome");

console.log(result);  // true
```

> In this example, the word `"awesome"` is found within the string `"JavaScript is awesome!"`, so the `includes()` method returns `true`.

Reviewer: ```suggestion
In this example, the word `awesome` is found within the string `JavaScript is awesome!`, so the `includes()` method returns `true`.
```

> It’s important to note that the `includes()` method is case-sensitive. This means that the exact match of the characters is required, including their case.

Reviewer: ```suggestion
It's important to note that the `includes()` method is case-sensitive. This means that the exact match of the characters is required, including their case.
```

For example:

```js
let phrase = "JavaScript is awesome!";
let result = phrase.includes("Awesome");

console.log(result);  // Output: false
```

> Since `"Awesome"` (with an uppercase "A") does not match `"awesome"` (with a lowercase "a"), the result is `false`.

Reviewer: ```suggestion
Since `Awesome` (with an uppercase `A`) does not match `awesome` (with a lowercase `a`), the result is `false`.
```

You can also use the `includes()` method to check for a substring starting at a specific index in the string by providing a second parameter:

```js
let text = "Hello, JavaScript world!";
let result = text.includes("JavaScript", 7);

console.log(result);  // true
```

> Here, the search for the substring `"JavaScript"` starts from the 7th position in the string, ensuring it skips any characters before this position.

Reviewer: ```suggestion
Here, the search for the substring `JavaScript` starts from the 7th position in the string, ensuring it skips any characters before this position.
```
